### PR TITLE
Use Uint8Array instead of Buffer in src/protobuf

### PR DESCRIPTION
**Pull request by jprochazk**
This should allow the `src/protobuf` to be lifted out of `masterchat` and put into its own library for easy consumption elsewhere (including in browsers)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "version": "0.12.0",
   "author": "Yasuaki Uechi <y@uechi.io> (https://uechi.io/)",
   "scripts": {
-    "build": "cross-env NODE_ENV=production rollup -c && rm -rf lib/{modules,protobuf} lib/{auth,errors,constants,index,pool,utils}.d.ts",
+    "build": "cross-env NODE_ENV=production rollup -c && shx rm -rf lib/{modules,protobuf} lib/{auth,errors,constants,index,pool,utils}.d.ts",
```
Comment (jprochazk) [inline package.json:7]: Fixed the typo here

File: src/protobuf/reader.ts [hunk 2]
```diff
@@ -1,41 +1,41 @@
...
-  eat(bytes: number): Buffer | null {
+  eat(bytes: number): Uint8Array | null {
     if (this.isEnded()) return null;
     return this.buf.slice(this.c, (this.c += bytes));
   }
 
   eatUInt32(): number | null {
     if (this.isEnded()) return null;
-    const n = this.buf.readUInt32LE(this.c);
+    const n = new DataView(this.buf).getUint32(this.c, true);
```
Comment (jprochazk) [inline src/protobuf/reader.ts:29]: This is super inefficient, but I'd like to keep it this way for simplicity sake (it's a 1:1 translation of what it was doing with `Buffer` before). Same in `eatUInt64`

File: src/protobuf/util.test.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { hextou8, u8tohex, b64tou8, u8tob64, concatu8 } from "./util";
```
Comment (jprochazk) [inline src/protobuf/util.test.ts:1]: Added tests for all the new utils

File: src/protobuf/util.ts [hunk 4]
```diff
@@ -50,16 +49,56 @@ export function toJSON(tokens: PBToken[]): string {
...
+  let out = "";
+  for (let i = 0; i < data.length; ++i) {
+    out += data[i].toString(16).padStart(2, "0");
+  }
+  return out;
+}
+
+const _atob = globalThis.atob as ((data: string) => string) | undefined;
+const _btoa = globalThis.btoa as ((data: string) => string) | undefined;
+export const b64tou8 = _atob
```
Comment (jprochazk) [inline src/protobuf/util.ts:96]: These try to use `atob`/`btoa` in browsers, and fall back to `Buffer` in node.js, where those functions are not defined. `atob`/`btoa` has issues with handling non-ascii text, but this is not a problem for us (as far as I can tell)

**Comment by uetchy** [comment]
Screen Shot 2021-10-04 at 10 26 55 PM
All tests passed! Thank you for the awesome job 🎉

**Pull request merged**